reactos/drivers/usb/usbport/endpoint.c

```diff
@@ -0,0 +1,1581 @@
+                {
+                    DPRINT1("USBPORT_DmaEndpointActive: FIXME call USBPORT_FlushIsoTransfer\n");
+                    ASSERT(FALSE); //USBPORT_FlushIsoTransfer();
+                }
+                else
+                {
+                    Transfer->CompletedTransferLen = CompletedLen;
+                }
+            }
+
+            RemoveEntryList(&Transfer->TransferLink);
+            Entry = Transfer->TransferLink.Flink;
+
+            if (Transfer->Flags & TRANSFER_FLAG_SPLITED)
+            {
+                DPRINT1("USBPORT_DmaEndpointActive: FIXME call USBPORT_CancelSplitTransfer\n");
+                ASSERT(FALSE); //USBPORT_CancelSplitTransfer();
+            }
+            else
+            {
+                InsertTailList(&Endpoint->CancelList, &Transfer->TransferLink);
+            }
+        }
+        else
+        {
+            Entry = Transfer->TransferLink.Flink;
+        }
+    }
+
+    return USBPORT_ENDPOINT_ACTIVE;
+}
+
+ULONG
+NTAPI
+USBPORT_DmaEndpointActive(IN PDEVICE_OBJECT FdoDevice,
+                          IN PUSBPORT_ENDPOINT Endpoint)
+{
+    PUSBPORT_DEVICE_EXTENSION FdoExtension;
+    PUSBPORT_REGISTRATION_PACKET Packet;
+    PLIST_ENTRY Entry;
+    PUSBPORT_TRANSFER Transfer;
+    LARGE_INTEGER TimeOut = {{0, 0}};
+    UCHAR CF;
+    MPSTATUS MpStatus;
+    KIRQL OldIrql;
+
+    DPRINT_CORE("USBPORT_DmaEndpointActive \n");
+
+    FdoExtension = (PUSBPORT_DEVICE_EXTENSION)FdoDevice->DeviceExtension;
+
+    Entry = Endpoint->TransferList.Flink;
+
+    if (Entry != &Endpoint->TransferList)
+    {
+        while (Entry && Entry != &Endpoint->TransferList)
+        {
+            Transfer = CONTAINING_RECORD(Entry,
+                                         USBPORT_TRANSFER,
+                                         TransferLink);
+
+            if (Transfer->Flags & 0x800)
+            {
+                USBPORT_QueueDoneTransfer(Transfer, USBD_STATUS_SUCCESS);
+                return USBPORT_ENDPOINT_ACTIVE;
+            }
+
+            if (!(Transfer->Flags & TRANSFER_FLAG_SUBMITED) &&
+                 !(Endpoint->Flags & ENDPOINT_FLAG_NUKE))
+            {
+                KeAcquireSpinLock(&FdoExtension->MiniportSpinLock, &OldIrql);
+
+                Packet = &FdoExtension->MiniPortInterface->Packet;
+
+                if (Transfer->Flags & TRANSFER_FLAG_ISO)
+                {
+                    DPRINT1("USBPORT_DmaEndpointActive: FIXME call SubmitIsoTransfer\n");
+
+                    MpStatus = Packet->SubmitIsoTransfer(FdoExtension->MiniPortExt,
+                                                         (PVOID)((ULONG_PTR)Endpoint + sizeof(USBPORT_ENDPOINT)),
+                                                         &Transfer->TransferParameters,
+                                                         Transfer->MiniportTransfer,
+                                                         NULL);//&Transfer->IsoTransferParameters);
```

Comment:
There will be a pointer to the parameter block for ISO transfer (in struct _USBPORT_TRANSFER).